crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -3150,3 +3150,11 @@ impl<TX: DbTx + 'static, N: NodeTypes + 'static> DBProvider for DatabaseProvider
         self.prune_modes_ref()
     }
 }
+
+impl<TX: DbTx, N: NodeTypes> DatabaseRef for DatabaseProvider<TX, N> {
```

Comment:
we now essentially have duplicate implementations for `DatabaseProvider` https://github.com/paradigmxyz/reth/blob/e78216d1807db79232e0efb1460a493cffbc811a/crates/storage/provider/src/providers/database/provider.rs#L524-L527

should we instead merge them into one that's using a trait?